modules/programs/neovim.nix

```diff
@@ -277,6 +285,7 @@ in {
       plugins = cfg.plugins
         ++ optionals cfg.coc.enable [ pkgs.vimPlugins.coc-nvim ];
       customRC = cfg.extraConfig;
+      initExtra = cfg.initExtra;
```

Comment:
https://github.com/NixOS/nixpkgs/blob/151c2f5a20c8d03a168242af18225f46786ef7c1/pkgs/applications/editors/neovim/utils.nix#L21-L42

This is not supported by the upstream function.

The only way to have configuration before the plugins section is to set `configure.beforePlugins`.

https://github.com/NixOS/nixpkgs/blob/151c2f5a20c8d03a168242af18225f46786ef7c1/pkgs/misc/vim-plugins/vim-utils.nix#L220-L230

Reply:
Looking at it, it shouldn't work I guess. But I am currently using my fork as home-manager in the flake and it achieves the behavior explained in the PR this way. Do you have any ideas on why do this happen?

Reply:
The attribute set passed to the function allows extra "params" for forward compatibility i.e. `{ attr1, attr2, ... }:`

Reply:
Can we just remove this parameter from `makeNeovimConfig` and use `cfg.initExtra` in the declaration of `generatedConfigViml`?
Also we might need to tweak the conditional in order to write the init.vim file. Currently if only initExtra is provided, config will not be written.